tests/delete/models.py

```diff
@@ -241,3 +241,20 @@ class GenericDeleteBottomParent(models.Model):
     generic_delete_bottom = models.ForeignKey(
         GenericDeleteBottom, on_delete=models.CASCADE
     )
+
+
+class Foo(models.Model):
+    pass
+
+
+class Bar(Foo):
+    pass
+
+
+class Baz(Foo):
+    class Meta:
+        proxy = True
+
+
+class BazItem(models.Model):
+    baz = models.ForeignKey("Baz", on_delete=models.CASCADE)
```

Comment:
Please try to reuse the existing models as much as possible when writing new tests as adding three tables per bugfix quickly adds up and slows down the suite initialization.

You could create a `ChildProxy(Child)` and adjust one of the models to have a `ForeignKey(ChildProxy)`.

These models also happen to be be self-documenting so the test should be easier to grok than when generic names like `Foo`, `Bar`, `Baz` are used.

Reply:
@charettes Thank you for the great feedback! When I tried it with Child and Parent, I noticed that it affected query counting in other tests, which required additional changes. So instead, I reused more appropriate models to set up the test.